test/widgets/message_list_test.dart

```diff
@@ -922,6 +922,37 @@ void main() {
         await tester.pump();
         tester.widget(find.text('new stream name'));
       });
+
+      testWidgets('does not navigate on tapping topic in TopicNarrow', (tester) async {
```

Comment:
Please also add a test before this, if there isn't one already, that it _does_ navigate if you tap the header in an interleaved view (i.e. a channel narrow or the combined feed, not a topic narrow), to check that the implementation doesn't accidentally remove the tap-to-navigate interaction there.

Similarly for the DM-message case, below. (DMs can appear in the combined feed.)